rclcpp/src/rclcpp/node.cpp

```diff
@@ -271,17 +271,21 @@ Node::list_parameters(
 
   // TODO(esteve): define parameter separator, use "." for now
   for (auto & kv : parameters_) {
-    if (std::any_of(prefixes.cbegin(), prefixes.cend(), [&kv, &depth](const std::string & prefix) {
+    if (((prefixes.size() == 0) &&
+      ((depth == 0) ||
```

Comment:
dfd3bf8

Matching change in https://github.com/ros2/system_tests/pull/140/commits/3d1d67219b039d5a0146cfa47f4d6ce2e8118401